app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -30,6 +30,42 @@ def supports_update?
       true
     end
 
+    def supports_operating_systems?
+      client.respond_to?(:operating_systems) || client.send(:client).respond_to?(:operating_systems)
+    end
+
+    def determine_os_type(host)
```

Comment:
Can you move the part that tries to match a OS with an oVirt OS (this whole file diff except line 177 and filtered_api stuff) to its own concern? It's really separate in behavior from the oVirt model.